### PR TITLE
Spawn actor entities with component authority based on lb strat if zo…

### PR DESCRIPTION
**Pull request by alastairdglennie**
### Description
When spawning Actor entities with load balancing enabled, we can skip an auth reassignment by directly assigning EntityACL write mapping and authority intent

**Before: Actor entities are spawned with UnrealWorker attribute then immediately after authority intent is set correctly and enforcer moves it (each Cube has OnAuthorityGained called twice)**

image

**After: Actor entity EntityACLs are set to the lb strategized worker on spawn, so there's no need to immediately load balance on spawn (each Cube has OnAuthorityGained called once)**

image

### CODE REVIEW

**Comment by KeyMaster-** [comment]
Note that this is dependent on #1768

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp [hunk 1]
```diff
@@ -221,7 +239,7 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
 
 	if (SpatialSettings->bEnableUnrealLoadBalancer)
 	{
-		ComponentDatas.Add(AuthorityIntent::CreateAuthorityIntentData(NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId()));
+		ComponentDatas.Add(AuthorityIntent::CreateAuthorityIntentData(IntendedVirtualWorkerId));
```
Comment (KeyMaster-) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp]: If the load balancing strategy returns an invalid virtual worker ID, then the ID set on the component will always be invalid - which wasn't possible before it seems (unless this code could have run before the worker got its own virtual worker ID). Maybe it's worth falling back on the local worker's virtual ID if the LB strategy didn't return a valid ID so the entity doesn't get stuck in limbo?

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp [hunk 2]
```diff
@@ -66,8 +75,24 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
 		AnyServerRequirementSet.Add(ZoningAttributeSet);
 		AnyServerOrClientRequirementSet.Add(ZoningAttributeSet);
 		AnyServerOrOwningClientRequirementSet.Add(ZoningAttributeSet);
+
+		check(NetDriver->LoadBalanceStrategy != nullptr);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp]: I assume we have more verbose errors than this, before this point, if LBS is nullptr?

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp [hunk 3]
```diff
@@ -66,8 +75,24 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
...
+		IntendedVirtualWorkerId = NetDriver->LoadBalanceStrategy->WhoShouldHaveAuthority(*Actor);
+		if (IntendedVirtualWorkerId == SpatialConstants::INVALID_VIRTUAL_WORKER_ID)
+		{
+			UE_LOG(LogEntityFactory, Error, TEXT("Load balancing strategy provided invalid virtual worker ID to spawn Actor. Actor: %s"), *Actor->GetName());
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp]: Output the LBS name also?

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp (hunk 2 above)
Comment (alastairdglennie) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp]: Currently, our netdriver logic makes it impossible for you to enable the load balancer without something being assigned.

Comment (jennifer-at-improbable-io) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp]: The NetDriver initialization should set and check it, and that should always be called before entity creation, so this should be fine to be a check.

**Pull request merged**